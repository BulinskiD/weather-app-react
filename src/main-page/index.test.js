import React from 'react';
import ReactDOM from 'react-dom';
import MainPage from './index';
import {shallow} from "enzyme";


describe('Main page', ()=>{

    const addFn = jest.fn();
    const removeFn = jest.fn();
    const cities = [{id: 1, name: "Sth", temperature: "20.0" }];

    beforeEach(()=>{
        addFn.mockClear();
        removeFn.mockClear();
    });

    it('should render correctly with given props', ()=>{
        const component = shallow(<MainPage onRemoveCity={removeFn} onAddCity={addFn} cities={cities} />);
        expect(component).toMatchSnapshot();
    });
});
