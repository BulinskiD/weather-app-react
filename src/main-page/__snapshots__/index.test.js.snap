// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page should render correctly with given props 1`] = `
<Fragment>
  <Online
    polling={true}
    wrapperType="span"
  >
    <CityAdder
      onAddCity={[MockFunction]}
    />
  </Online>
  <Offline
    polling={true}
    wrapperType="span"
  >
    <Alert
      closeLabel="Close alert"
      show={true}
      transition={[Function]}
      variant="warning"
    >
      Dodawanie nowych miast niemo≈ºliwe w trybie offline!
    </Alert>
  </Offline>
  <hr />
  <ForecastTable
    cities={
      Array [
        Object {
          "id": 1,
          "name": "Sth",
          "temperature": "20.0",
        },
      ]
    }
    onRemoveCity={[MockFunction]}
  />
</Fragment>
`;
